### DISCUSSION TITLE
Retrieve custom field value from netbox shell

### DISCUSSION THREAD

**Discussion by tux-mania** [netbox-community/netbox#9461] [Q&A] [answered]
Hi all.

Getting stuck again with custom_fields here.
I need to retrieve the value for custom_field 'LAN' in Devices.
But I do not know how to get it. Doing a `dir(Device)` I see the properties `cf` and `custom_field_data`. But I don't know how to use it to extract the data needed.

Could you please help?
Thanks.

**Comment by ben-marquardt-tfs** [comment, marked as answer]
You should be able to find your custom field of 'Lan' with custom_field_data['LAN']. From what you have said it seems like you would just do Device.custom_field_data['LAN']

**Reply by tux-mania** [reply]
I think (sure) I'm misunderstanding the way it all works.
When I try to do the command you gave me, I get:
```
>>> Device.custom_field_data['LAN']
Traceback (most recent call last):
  File "<console>", line 1, in <module>
TypeError: 'DeferredAttribute' object is not subscriptable
```
Should I access the data in another way? A bit lost with the objects thing...

**Reply by tux-mania** [reply]
Sorry for the previous answer. After checking again the official documentation, I found the correct way to do it.
So, using your command and the documentation, it works with:
```
>>> for device in Device.objects.all():
...     print(device.custom_field_data['LAN'])
```
Thanks for your help!
